test/token/ERC20WithPermit.test.js

```diff
@@ -0,0 +1,1069 @@
+              allowance,
+              anotherAccount.address,
+              deadline
+            )
+
+            await token
+              .connect(anotherAccount)
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                allowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+
+            expect(
+              await token.allowance(
+                permittingHolder.address,
+                anotherAccount.address
+              )
+            ).to.equal(allowance)
+          })
+        })
+
+        context("when the spender had an approved amount", () => {
+          beforeEach(async () => {
+            const deadline = tomorrow
+            const initialAllowance = allowance.sub(10)
+            const signature = await getApproval(
+              initialAllowance,
+              anotherAccount.address,
+              deadline
+            )
+
+            await token
+              .connect(anotherAccount)
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                initialAllowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+          })
+
+          it("should approve the requested amount and replaces the previous one", async () => {
+            const deadline = tomorrow
+            const signature = await getApproval(
+              allowance,
+              anotherAccount.address,
+              deadline
+            )
+
+            await token
+              .connect(anotherAccount)
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                allowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+
+            expect(
+              await token.allowance(
+                permittingHolder.address,
+                anotherAccount.address
+              )
+            ).to.equal(allowance)
+          })
+        })
+      })
+    })
+
+    context("when the spender is the zero address", () => {
+      const allowance = permittingHolderBalance
+      it("should revert", async () => {
+        const deadline = tomorrow
+        const signature = await getApproval(allowance, ZERO_ADDRESS, deadline)
+
+        await expect(
+          token
+            .connect(anotherAccount)
+            .permit(
+              permittingHolder.address,
+              ZERO_ADDRESS,
+              allowance,
+              deadline,
+              signature.v,
+              signature.r,
+              signature.s
+            )
+        ).to.be.revertedWith("Approve to the zero address")
+      })
+    })
+
+    context("when given never expiring permit", () => {
+      // uint(-1)
```

Comment:
Is this still valid in solidity 0.8? We use max uint256 in the code, so having `uint(-1)` here is confusing.

Reply:
`uint(-1) = type(uint256).max`, see [Solidity 8 breaking changes](https://docs.soliditylang.org/en/v0.8.0/080-breaking-changes.html):

> In particular, replace usages of uint(-1) with type(uint).max.

I left it as uint(-1) because it's how it's referenced in the EIP.

Reply:
If we want to keep referencing code and EIP we should mention them both, something like:
```suggestion
      // uint(-1) = type(uint256).max
```

Reply:
https://github.com/thesis/solidity-contracts/pull/3/commits/ee42c58081a345f759f8f89b7022c0f70109f14d